File: 1071-binary-prefix-divisible-by-5/binary-prefix-divisible-by-5.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        int prefix = 0;
        vector<bool> list;
        for(auto& n : nums){
            prefix = ((prefix << 1) + n) % 5;
            list.push_back(prefix == 0);
        }
        return list;
    }
};

File: 1071-binary-prefix-divisible-by-5/binary-prefix-divisible-by-5.java
class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> result = new ArrayList<>();
        int prefix = 0;
        int n = nums.length;
        for(int i=0;i<n;i++){
            prefix = ((prefix << 1) + nums[i]) % 5;
            result.add(prefix == 0);
        }
        return result;
    }
}